### PR TITLE
Switch to 3 column display of implementation statements

### PR DESCRIPTION
**Pull request by gregelin**
## About
This PR replaces the accordion-based presentation of component control implementation statements with an initial implementation of @kerryrm's better designed 3 Column layout.

The "3-Column" statement layout stacks all control statements on top of each other with the content of control implementation statements aligned in the middle column making it possible to read the control implementation statements as one long combined statement.

This style was applied to:
- Component library component detail page
- System selected components component detail page
- System selected controls control detail page

## 3 Column statement layout
Screen Shot 2021-02-24 at 8 30 43 AM

## Original accordion layout

Screen Shot 2021-02-24 at 8 33 09 AM

### CODE REVIEW

File: templates/components/element_details_tabs_smtform.html [hunk 1]
```diff
@@ -64,7 +64,10 @@ <h4 id="panel-panel_num-title" class="panel-title">
                         <button type="button" name="save" value="save" class="btn btn-xs btn-success" onclick="save_smt('smt_panel_num');return false;">Save</button>
 
                     </div>
-                    <div id="document-panel_num-body" class="panel-collapse collapse {% if forloop.first %}in{% endif %}" role="tabpanel" aria-labelledby="document-panel_num-title">
+
+
+asdfdfsfdfd
```
Comment (davidpofo) [inline templates/components/element_details_tabs_smtform.html:69]: no. Pete said is there more to do here?

File: templates/systems/element_detail_tabs.html [hunk 2]
```diff
@@ -28,10 +29,12 @@
   }
 
   #above-tab-content, #tab-content {
-    max-width: 950px;
+    max-width: 1250px;
     margin: auto;
   }
 
+  .container { width:90%;}
```
Comment (davidpofo) [inline templates/systems/element_detail_tabs.html:36]:
```suggestion
  .container { width: 90%; }
```

File: templates/systems/element_detail_tabs.html [hunk 3]
```diff
@@ -81,88 +84,109 @@
...
                       {% for smt in impl_smts %}
-                          <div id="panel-{{ forloop.counter }}" class="panel panel-default">
+                          <div id="panel-{{ forloop.counter }}" class="">
                               <div class="panel-heading" role="tab" id="document-{{ forloop.counter }}-title">
+
+                                <div class="row statement-text">
+                                  <div id="" class="col-xs-3 col-sm-3 col-md-3 col-lg-3 col-xl-3">
+                                    <span id="producer_element-panel_num-control" class="control-id-text">{{ smt.sid|upper }} {% if smt.pid %}{{ smt.pid }}.{% endif %}</span>
+                                  </div>
+                                  <div id="" class="col-xs-6 col-sm-6 col-md-6 col-lg-6 col-xl-6 statement-text-block">
```
Comment (davidpofo) [inline templates/systems/element_detail_tabs.html:161]: div id please

File: templates/systems/element_detail_tabs.html [hunk 4]
```diff
@@ -81,88 +84,109 @@
...
+                                    {{ smt.body }}
+                                  </div>
+                                  <div id="" class="col-xs-3 col-sm-3 col-md-3 col-lg-3 col-xl-3">
```
Comment (davidpofo) [inline templates/systems/element_detail_tabs.html:164]: div id please

File: templates/systems/element_detail_tabs.html [hunk 5]
```diff
@@ -81,88 +84,109 @@
...
+                                    <a role="button" class="" data-toggle="collapse" data-parent="#accordion" href="#document-{{ forloop.counter }}-body" aria-expanded="false" aria-controls="document-{{ forloop.counter }}-body">edit</a>
+                                    <br />
+                                    {{ smt.remarks }}
+                                  </div>
+                                </div>
+
                                   <h4 id="panel-{{ forloop.counter }}-title" class="panel-title">
-                                      <a role="button" data-toggle="collapse" data-parent="#accordion" href="#document-{{ forloop.counter }}-body" aria-expanded="false" aria-controls="document-{{ forloop.counter }}-body">
+                                      <!-- <a role="button" data-toggle="collapse" data-parent="#accordion" href="#document-{{ forloop.counter }}-body" aria-expanded="false" aria-controls="document-{{ forloop.counter }}-body">
```
Comment (davidpofo) [inline templates/systems/element_detail_tabs.html:172]:
```suggestion
```
Extra uncommented lines are not useful here.

File: templates/systems/element_detail_tabs.html [hunk 6]
```diff
@@ -410,4 +422,10 @@ <h3><button type="submit" id="new_component_statement" class="small btn btn-md b
         }
       </script>
 
+      <script>
+        $( document ).ready(function() {
```
Comment (davidpofo) [inline templates/systems/element_detail_tabs.html:426]: This is probably where edit will get its functionality

**Review by davidpofo — COMMENTED** [review]
PR party 2/25 its there

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 GovReady-Q Release Notes
 ========================
 
-v999 (February XX, 2021)
+v0.9.1.52 (February 16, 2021)
```
Comment (davidpofo) [inline CHANGELOG.md:4]: This should not be changed in a branch to develop. Same for the VERSION file.

File: templates/components/element_details_tabs_smtform.html (hunk 1 above)
Comment (davidpofo) [reply templates/components/element_details_tabs_smtform.html:69]: @gregelin

File: templates/systems/element_detail_tabs.html (hunk 3 above)
Comment (davidpofo) [reply templates/systems/element_detail_tabs.html:161]: @gregelin

**Comment by gregelin** [comment]
The test that appears to be failing is `python manage.py test controls.tests.ControlComponentTests`. It appears to fail because after 4 component controls to a control, clicking on the newly added component tab opens the previous component statement block for editing instead of the newly created component statement block.

This incorrect accordion opening does not appear to happen when there are 3 new component statements. I'm not sure where the pointers or counters get lost.

**Comment by gregelin** [comment]
Interesting, the error seems to occur with panel/document 5. Two items end up with "5". Continuing to add new components correctly up counts to 6, 7, 8, etc.

Why would 5 get the same number twice?

**Comment by gregelin** [comment]
Resolved the failing test. The problem was I was counting elements with class "panel." In previous version of page the only "panel" class on the page were the accordion panels for the control implementation statement. However, when I added the tab layout the tabs also use the "panel" class. Turns out I was always getting a count of 4 objects with class panel on adding new panels. But it was only when there in fact 5 controls (0 - 4) did the number 5 repeat itself as an index.

Problem was corrected by giving the interior statement blocks their own class of "smt-panel" and counting entities with class "smt-panel" to get the correct count.

File: templates/systems/element_detail_tabs.html (hunk 3 above)
Comment (gregelin) [reply templates/systems/element_detail_tabs.html:161]: blank IDs removed.

File: templates/systems/element_detail_tabs.html (hunk 6 above)
Comment (gregelin) [reply templates/systems/element_detail_tabs.html:421]: Not sure what you mean by the comment.

**Pull request merged**